hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/ExpressionPredicates.java

```diff
@@ -223,7 +228,98 @@ public ColumnPredicate bindValueLiteral(ValueLiteralExpression valueLiteral) {
 
     @Override
     public FilterPredicate filter() {
-      return toParquetPredicate(getFunctionDefinition(), literalType, columnName, literal);
+      Serializable convertedLiteral = convertImplicitly(literalType, literal);
+      return toParquetPredicate(getFunctionDefinition(), literalType, columnName, convertedLiteral);
+    }
+
+    private Serializable convertImplicitly(LogicalType literalType, Serializable literal) {
+      try {
```

Comment:
Implicit type conversion is a intricate topic, I would suggest we move this logic into a separate class `ImplicitTypeConversions` first, when Flink supports that, we can deprecate it.

Reply:
OK. I'll do it.